### ISSUE TITLE
[red-knot] Strange error emitted on variable annotated with `type[EllipsisType]`

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ruff#15271]
I feel like this should be fine:

```py
from types import EllipsisType

some_variable: type[EllipsisType] = type(...)
```

But if I run `cargo run -p red_knot --project ../experiment --python-version="3.10"`, red-knot complains:

```
error[lint:invalid-assignment] /Users/alexw/dev/experiment/foo.py:3:1 Implicit shadowing of class `EllipsisType`; annotate to make it explicit if this is intentional
```

I'm not sure what exactly red-knot thinks I'm shadowing here 😄 The variable must be explicitly annotated for the error to occur.

**Comment by AlexWaygood** [comment]
The error also goes away if I make this change:

```diff
  from types import EllipsisType

- some_variable: type[EllipsisType] = type(...)
+ some_variable: type[EllipsisType] = EllipsisType
```

**Comment by AlexWaygood** [comment]
Oh, shoot, this doesn't repro on `main`, only on one of my PR branches.

...And I think it's "correct" (with our current logic) to emit an error there anyway, because we don't yet apply any special casing to `type()` -- the inferred type of `type(...)` is probably just `Instance(<builtins.type>)`.

Thanks!
